core/src/main/java/com/oath/oak/OakUnsafeRef.java

```diff
@@ -0,0 +1,28 @@
+package com.oath.oak;
+
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+public interface OakUnsafeRef {
```

Comment:
Other concern: you disregard the OakR and OakW! Although they are similar (put aside transform) one should return read-only buffer and the other should return writable buffer... What about that?

Reply:
This is now noted inside the internal documentation of `OakUnsafeDirectBuffer` that depending on the context, `getByteBuffer()` might return a read-only buffer.